# Corelation

# Topic- Corelation between attrition and other variables

In order to find the interdependency of the variables DistanceFromHome,Education ,Gender,NumCompaniesWorked,PercentSalaryHike ,MonthlyIncome,YearsSinceLastPromotion,TotalWorkingYears,TrainingTimesLastYear, YearsAtCompany, YearsWithCurrManager ,Age from that of Attrition, we executed the Correlation Analysis as follows :

In [3]:
import pandas as pd
import numpy as np
dataset=pd.read_csv("general_data.csv")
dataset.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [22]:
dataset.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   
4409  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   
4409      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   
4409               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   
4409             False               True                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4405           False                    False                 False  
4406           False    

In [33]:
dataset.dropna()

Age  Attrition     BusinessTravel              Department  \
0      51          0      Travel_Rarely                   Sales   
1      31          1  Travel_Frequently  Research & Development   
2      32          0  Travel_Frequently  Research & Development   
3      38          0         Non-Travel  Research & Development   
4      32          0      Travel_Rarely  Research & Development   
...   ...        ...                ...                     ...   
4404   29          0      Travel_Rarely                   Sales   
4405   42          0      Travel_Rarely  Research & Development   
4406   29          0      Travel_Rarely  Research & Development   
4407   25          0      Travel_Rarely  Research & Development   
4408   42          0      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0          0  ...                 1.0      Y                11              8   
1          0  ...                 0.0      Y                23              8   
2          1  ...                 1.0      Y                15              8   
3          1  ...                 3.0      Y                11              8   
4          1  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404       0  ...                 1.0      Y                18              8   
4405       0  ...                 3.0      Y                17              8   
4406       1  ...                 2.0      Y                15              8   
4407       1  ...                 0.0      Y                20              8   
4408       1  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...         

In [26]:
dataset.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4405    False
4406    False
4407    False
4408    False
4409    False
Length: 4410, dtype: bool

In [27]:
dataset.drop_duplicates()

Age  Attrition     BusinessTravel              Department  \
0      51          0      Travel_Rarely                   Sales   
1      31          1  Travel_Frequently  Research & Development   
2      32          0  Travel_Frequently  Research & Development   
3      38          0         Non-Travel  Research & Development   
4      32          0      Travel_Rarely  Research & Development   
...   ...        ...                ...                     ...   
4405   42          0      Travel_Rarely  Research & Development   
4406   29          0      Travel_Rarely  Research & Development   
4407   25          0      Travel_Rarely  Research & Development   
4408   42          0      Travel_Rarely                   Sales   
4409   40          0      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0          0  ...                 1.0      Y                11              8   
1          0  ...                 0.0      Y                23              8   
2          1  ...                 1.0      Y                15              8   
3          1  ...                 3.0      Y                11              8   
4          1  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405       0  ...                 3.0      Y                17              8   
4406       1  ...                 2.0      Y                15              8   
4407       1  ...                 0.0      Y                20              8   
4408       1  ...                 0.0      Y                14              8   
4409       1  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...         

In [4]:
from sklearn import preprocessing
Label_encoder=preprocessing.LabelBinarizer()
dataset["Attrition"]=Label_encoder.fit_transform(dataset["Attrition"])
from scipy.stats import pearsonr
stats,p=pearsonr(dataset.Attrition,dataset.Age)
print(stats,p)

-0.159205006865775 1.996801615893625e-26


In [11]:
stats,p=pearsonr(dataset.Attrition,dataset.DistanceFromHome)
print(stats,p)

-0.009730141010179692 0.5182860428050771


In [15]:
stats,p=pearsonr(dataset.Attrition,dataset.Education)
print(stats,p)

-0.015111167710968813 0.3157293177118575


In [19]:
from sklearn import preprocessing
Label_encoder1=preprocessing.LabelBinarizer()
dataset["Gender"]=Label_encoder1.fit_transform(dataset["Gender"])
stats,p=pearsonr(dataset.Attrition,dataset.Gender)
print(stats,p)

0.018125078877010335 0.22881970951790567


In [29]:
stats,p=pearsonr(dataset.Attrition,dataset.PercentSalaryHike)
print(stats,p)

0.03253259489105355 0.030743386433355353


In [7]:
stats,p=pearsonr(dataset.Attrition,dataset.StockOptionLevel)
print(stats,p)

-0.006838852403261037 0.6498072937477383


In [9]:
stats,p=pearsonr(dataset.Attrition,dataset.TrainingTimesLastYear)
print(stats,p)

-0.049430576244253066 0.001024706191536548


In [10]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsAtCompany)
print(stats,p)

-0.13439221398997386 3.163883122493571e-19


In [11]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsSinceLastPromotion)
print(stats,p)

-0.03301877514258329 0.02833033618939086


In [12]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsWithCurrManager)
print(stats,p)

-0.1561993159016244 1.7339322652951965e-25


__The inference of the above analysis are as follows:__
    
Attrition & DistanceFromHome:
    
As r = -0.009, there’s low negative correlation between Attrition and DistanceFromHome
As the P value of 0.518 is > 0.05, we are accepting H0 and hence there’s no significant correlation
between Attrition & DistanceFromHome


Attrition & MonthlyIncome:
    
As r = -0.031, there’s low negative correlation between Attrition and MonthlyIncome
As the P value of 0.038 is < 0.05, we are accepting Ha and hence there’s significant correlation
between Attrition & MonthlyIncome


Attrition & TotalWorkingYears:
    
As r = -0.17, there’s low negative correlation between Attrition and TotalWorkingYears
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & TotalWorkingYears


Attrition & YearsAtCompany:
    
As r = -0.1343, there’s low negative correlation between Attrition and YearsAtCompany
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & YearsAtCompany


Attrition & YearsWithCurrManager:
    
As r = -0.1561, there’s low negative correlation between Attrition and YearsWithCurrManager
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & YearsWithCurrManager


Attrition & YearsSinceLastPromotion:
    
As r = -0.033, there’s low negative correlation between Attrition and YearsSinceLastPromotion
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & YearsSinceLastPromotion



Attrition & TrainingTimesLastYear:
    
As r = -0.0494, there’s low negative correlation between Attrition and TrainingTimesLastYear
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & TrainingTimesLastYear



Attrition & StockOptionLevel:
    
As r = -0.0068, there’s low negative correlation between Attrition and StockOptionLevel
As the P value is > 0.05, we are accepting H0 and hence there’s significant correlation between
Attrition & StockOptionLevel



Attrition & PercentSalaryHike:
    
As r = 0.0325, there’s low positive correlation between Attrition and PercentSalaryHike
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & PercentSalaryHike


Attrition & Gender:
    
As r = 0.0181, there’s low positive correlation between Attrition and Gender
As the P value is > 0.05, we are accepting H0 and hence there’s significant correlation between
Attrition & Gender


Attrition & Education:
    
As r = -0.0151, there’s low negative correlation between Attrition and Education
As the P value is > 0.05, we are accepting H0 and hence there’s significant correlation between
Attrition & Education


Attrition & Age:
    
As r = -0.1592, there’s low negative correlation between Attrition and Age
As the P value is < 0.05, we are accepting Ha and hence there’s significant correlation between
Attrition & Age